samples/extensions/full_screen_exclusive/full_screen_exclusive.cpp

```diff
@@ -0,0 +1,1201 @@
+						isFullScreenExclusive                                      = true;
+						LOGI("Exclusive Fullscreen Mode Detected!")
+					}
+					break;
+				default:        // FullscreenExclusiveEXT = Default
+					isRecreate = false;
+					break;
+			}
+
+			// now if to recreate the swapchain and everything related:
+			if (isRecreate)
+			{
+				FullScreenExclusive::recreate();
+			}
+		}
+	}
+#endif
+}
+
+void FullScreenExclusive::update_application_window()
+{
+#if defined(VK_USE_PLATFORM_WIN32_KHR)
+	if (application_window_status == ApplicationWindowMode::Fullscreen && isWindowed)        // check if it is already in fullscreen, if is, then do nothing
+	{
+		isWindowed = false;
+
+		GetWindowPlacement(HWND_applicationWindow, &wpc);
+		if (HWND_style == 0)
+			HWND_style = GetWindowLong(HWND_applicationWindow, GWL_STYLE);
+		if (HWND_style_previous == 0)
+			HWND_style_previous = GetWindowLong(HWND_applicationWindow, GWL_EXSTYLE);
+
+		long HWND_newStyle = HWND_style;
+		HWND_newStyle &= ~WS_BORDER;
+		HWND_newStyle &= ~WS_DLGFRAME;
+		HWND_newStyle &= ~WS_THICKFRAME;
+
+		long HWND_newStyle_previous = HWND_style_previous;
+		HWND_newStyle_previous &= ~WS_EX_WINDOWEDGE;
+
+		SetWindowLong(HWND_applicationWindow, GWL_STYLE, HWND_newStyle | static_cast<long>(WS_POPUP));
+		SetWindowLong(HWND_applicationWindow, GWL_EXSTYLE, HWND_newStyle_previous | WS_EX_TOPMOST);
+		ShowWindow(HWND_applicationWindow, SW_SHOWMAXIMIZED);
+	}
+	else if (application_window_status == ApplicationWindowMode::Windowed && !isWindowed)        // check if it is already "windowed", if is, then do nothing
+	{
+		isWindowed = true;
+
+		SetWindowLong(HWND_applicationWindow, GWL_STYLE, HWND_style);
+		SetWindowLong(HWND_applicationWindow, GWL_EXSTYLE, HWND_style_previous);
+		ShowWindow(HWND_applicationWindow, SW_SHOWNORMAL);
+		SetWindowPlacement(HWND_applicationWindow, &wpc);
+	}
+#endif
+}
+
+void FullScreenExclusive::recreate()
+{
+#if defined(VK_USE_PLATFORM_WIN32_KHR)
+	// Check if there IS a device, if not don't do anything
+	if (context.device != VK_NULL_HANDLE)
+	{
+		// Step: 0) Idle the device, destroy/teardown the current swapchain and frame buffers.
+		vkDeviceWaitIdle(context.device);        // pause the renderer
+		teardown_frame_buffers(context);         // basically destroy everything swapchain related
+
+		// Step: 1) recreate the swapchain with its properly selected FullscreenExclusive enum value
+		init_swapchain();
+
+		// Step: 2) recreate the frame buffers using the newly created swapchain
+		init_frame_buffers(context);
+
+		// Step: 3-1) update the window mode, corresponding to the FullscreenExclusive enum value
+		update_application_window();
+
+		// Step: 3-2) remember: ALWAYS change the application window mode BEFORE acquire the full screen exclusive mode!
+		if (isFullScreenExclusive)
+		{
+			VkResult result = vkAcquireFullScreenExclusiveModeEXT(context.device, context.swapchain);
+			if (result == VK_SUCCESS)
+			{
```

Comment:
`vkReleaseFullScreenExclusiveModeEXT` is not used anywhere in this sample. Could you elaborate on why?